### PR TITLE
US128215: new question analysis skeleton page

### PR DESCRIPTION
**Pull request by TarasShDev**
[US128215](https://rally1.rallydev.com/#/23525809118d/workviews?detail=%2Fuserstory%2F604748071016&view=801ed3fc-8416-445a-9696-22fd3d5e8a50&fdp=true?fdp=true): [AQ] New Question Analysis Page
This is a **skeleton page** to begin with.
Design [page](https://xd.adobe.com/view/21fb1c7a-c885-400a-a55b-887d2b26a237-74a8/screen/d026bdb1-c3c9-4208-ae6b-14abd185785a/).

FYI @Brightspace/draco-vis 

**Demo** 🎥 

https://user-images.githubusercontent.com/46570898/127311468-176ba62b-e614-4dff-9f02-4c5e046a1af2.mp4

image

### CODE REVIEW

File: assessment-quality-dashboard.js [hunk 1]
```diff
@@ -76,6 +78,7 @@ class AssessmentQualityDashboard extends SkeletonMixin(Localizer(MobxLitElement)
 	get navBackTitle() {
 		const route = this.currentRoute;
 		if (route === '' || route === '/') return this.localize('dashboard:backToInsightsPortal');
+		else if (route.includes('/question')) return this.localize('dashboard:backToQuiz', { quizName: 'Quiz' });
```
Comment (Vitalii-Misechko) [inline assessment-quality-dashboard.js:81]: You should use langterm instead of `Quiz` or proper quiz name

File: route-loader.js [hunk 2]
```diff
@@ -49,6 +49,13 @@ registerRoutes([
 							>
 							</d2l-insights-settings-view>`,
 		loader: () => import('./pages/settings-view.js')
+	},
+	{
+		pattern: '/quiz/:quizId/question/:questionId',
+		view: () => html`<d2l-insights-question-item-analysis-view
+							>
+							</d2l-insights-question-item-analysis-view>`,
```
Comment (Vitalii-Misechko) [inline route-loader.js:57]: Could you pass to the new page `questionId`, `serverData`, and `skeleton`?

**Comment by Vitalii-Misechko** [comment]
Could you add basic unit tests (constructor, axe, etc)?

File: route-loader.js (hunk 2 above)
Comment (TarasShDev) [reply route-loader.js:57]: Sure

File: route-loader.js [hunk 3]
```diff
@@ -49,6 +49,13 @@ registerRoutes([
 							>
 							</d2l-insights-settings-view>`,
 		loader: () => import('./pages/settings-view.js')
+	},
+	{
+		pattern: '/quiz/:quizId/question/:questionId',
+		view: () => html`<d2l-insights-question-item-analysis-view
```
Comment (NicholasHallman) [inline route-loader.js:55]: You can pass a ctx parameter into `view` which will give you access to `ctx.params.quizId` and `ctx.params.questionId` to access the route parameters

File: pages/question-item-analysis-view.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+					padding: 0.75rem 18px;
+				}
+			}`
+		];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-aria-loading-progress-placeholder"></div>
+            <div class="d2l-insights-settings-page-main-content">
```
Comment (NicholasHallman) [inline pages/question-item-analysis-view.js:97]: `nit` The indentation here is inconsistent

File: assessment-quality-dashboard.js (hunk 1 above)
Comment (TarasShDev) [reply assessment-quality-dashboard.js]: Updated to use `quiz name`.

File: route-loader.js (hunk 2 above)
Comment (TarasShDev) [reply route-loader.js:61]: Done

File: route-loader.js (hunk 3 above)
Comment (TarasShDev) [reply route-loader.js]: Yep, updated.

File: pages/question-item-analysis-view.js (hunk 4 above)
Comment (TarasShDev) [reply pages/question-item-analysis-view.js]: Updated

**Comment by TarasShDev** [comment]
> Could you add basic unit tests (constructor, axe, etc)?

Updated.

**Review by NicholasHallman — APPROVED** [review]

File: pages/quiz-item-analysis-view.js [hunk 5]
```diff
@@ -242,7 +242,8 @@ class QuizItemAnalysisView extends SkeletonMixin(Localizer(MobxLitElement)) {
 		const handleRedirectToQuestionDetail = (e) => {
 			const preId = 'question-tooltip-';
 			const quizItemId = e.target.id.substring(preId.length);
-			redirect(`/quiz/${this.quizId}/question/${quizItemId}`);
+			const quizName = this.serverData.quizItemData.quizName;
+			redirect(`/quiz/${this.quizId}/question/${quizItemId}?from=quiz,${quizName}`);
```
Comment (Vitalii-Misechko) [inline pages/quiz-item-analysis-view.js:246]: `curious` what if `quizName` contains Unicode symbols? Will it crash the app?

File: pages/question-item-analysis-view.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+			heading3Styles,
+			css`
+			:host {
+				display: block;
+				padding: 0 30px;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-heading-button-group {
```
Comment (Vitalii-Misechko) [inline pages/question-item-analysis-view.js:45]: There are a few unsued CSS styles. Could you remove them?

File: pages/question-item-analysis-view.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+import '@brightspace-ui-labs/ou-filter/ou-filter';
+import '@brightspace-ui-labs/pagination/pagination';
+
+import { bodyCompactStyles, bodySmallStyles, bodyStandardStyles, heading3Styles } from '@brightspace-ui/core/components/typography/styles';
+import { css, html } from 'lit-element';
+import { Localizer } from '../locales/localizer';
+import { MobxLitElement } from '@adobe/lit-mobx';
+import { SkeletonMixin } from '@brightspace-ui/core/components/skeleton/skeleton-mixin.js';
+
+class QuestionItemAnalysisView extends SkeletonMixin(Localizer(MobxLitElement)) {
```
Comment (Vitalii-Misechko) [inline pages/question-item-analysis-view.js:15]: We don't need `MobxLitElement` ancestor. `LitElement` is enough so far

File: test/pages/question-item-analysis-view.test.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+			runConstructor('d2l-insights-question-item-analysis-view');
+		});
+	});
+
+	describe('render', () => {
+		it('should show header', async() => {
+			const el = await fixture(html`<d2l-insights-question-item-analysis-view
+            ></d2l-insights-question-item-analysis-view>`);
+
+			await el.renderRoot.querySelectorAll('div')[1].updateComplete;
```
Comment (Vitalii-Misechko) [inline test/pages/question-item-analysis-view.test.js:26]: this line doesn't make sense. `updateComplete` is a promise and[ a part ](https://lit-element.polymer-project.org/guide/lifecycle#updatecomplete)of `LitElement` framework. `div` should not have this promise

File: test/visual-diff/assessment-quality-dashboard.visual-diff.js [hunk 9]
```diff
@@ -45,7 +45,7 @@ describe('assessment-quality-dashboard', () => {
 			deviceScaleFactor: 1
 		});
 		const rect = await visualDiff.getRect(page, 'body');
-		await new Promise(resolve => setTimeout(resolve, 300));
+		await new Promise(resolve => setTimeout(resolve, 3000));
```
Comment (Vitalii-Misechko) [inline test/visual-diff/assessment-quality-dashboard.visual-diff.js:48]: Does this help to reduce visual-diff flakiness? If yes could you try to apply a smaller number if possible

File: pages/quiz-item-analysis-view.js (hunk 5 above)
Comment (TarasShDev) [reply pages/quiz-item-analysis-view.js:246]: **No** 😂😅
image
image

File: pages/question-item-analysis-view.js (hunk 6 above)
Comment (TarasShDev) [reply pages/question-item-analysis-view.js]: Sure.

File: pages/question-item-analysis-view.js (hunk 7 above)
Comment (TarasShDev) [reply pages/question-item-analysis-view.js]: Updated.

File: test/pages/question-item-analysis-view.test.js (hunk 8 above)
Comment (TarasShDev) [reply test/pages/question-item-analysis-view.test.js]: Removed.

File: test/visual-diff/assessment-quality-dashboard.visual-diff.js (hunk 9 above)
Comment (TarasShDev) [reply test/visual-diff/assessment-quality-dashboard.visual-diff.js:48]: I've tried 500 and 1500 - test couldn't pass.
Took [quiz-item-analysis-view](https://github.com/Brightspace/insights-assessment-quality-dashboard/blob/master/test/visual-diff/quiz-item-analysis-view.visual-diff.js#L51) as an example.

**Review by Vitalii-Misechko — APPROVED** [review]
LGTM!

**Pull request merged**